src/main/java/Duke/Commands/EventCommand.java

```diff
@@ -0,0 +1,123 @@
+        Event e;
+        try{
+            LocalDate parsedDate = stringToLocalDate(string);
+            LocalDate endDate = stringToLocalDate(end);
+            if(parsedDate.isAfter(endDate)){
+                throw new EventException(false, false, true, false);
+            }
+            e = new Event(name, parsedDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy")),
+                    endDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy")));
+        }catch (EventException event){
+            throw new EventException(false, false, true, false);
+        }catch (DateTimeException d) {
+            try {
+                LocalDateTime parsedDate = stringToLocalDateTime(string);
+                LocalDateTime endDate = stringToLocalDateTime(end);
+                if(parsedDate.isAfter(endDate)){
+                    throw new EventException(false, false, true, false);
+                }
+                e = new Event(name, parsedDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy, HH:mm")),
+                        endDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy, HH:mm")));
+            } catch (EventException event){
+                throw new EventException(false, false, true, false);
+            }
+            catch (DateTimeException g) {
+                try {
+                    LocalTime parsedDate = stringToLocalTime(string);
+                    LocalTime endDate = stringToLocalTime(end);
+                    if(parsedDate.isAfter(endDate)){
+                        throw new EventException(false, false, true, false);
+                    }
+                    e = new Event(name, parsedDate.format(DateTimeFormatter.ofPattern("HH:mm")),
+                            endDate.format(DateTimeFormatter.ofPattern("HH:mm")));
+                }catch (EventException y){
+                    throw new EventException(false, false, true, false);
+                }catch (DateTimeException z) {
+                    throw new EventException(false, false, false, true);
+                }
+            }
+        }
+        return e;
+    }
+
+    /**
+     * Adds Event task or handle exceptions
+     *
+     * @param tasks to change the taskList if necessary when no error
+     * @param ui
+     * @param storage to change the file in the if necessary when no error
+     * @return String returns the string of the output that informs the action has been complete.
+     * @throws DukeException if there no description after Event no time or time is wrong format
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (commandDescription.length() == 5 || commandDescription.length() == 6) {
+            throw new EventException(true, false, false, false);
+        }
+        String s = "";
+        int index = -1;
+        int end = -1;
+        boolean duration = false;
+        boolean time = false;
```

Comment:
Perhaps you can use a more descriptive variable name to increase readability :)

Reply:
Thank you for your comment, yes I agree that my variable name is bad. Would it be better if my variable name is durationPresent and timePresent?

Reply:
It's okay, my variable naming is also bad sometimes ._. Yes, it's better! 👍